frontend/packages/console-shared/src/utils/resource-utils.ts

```diff
@@ -244,6 +245,22 @@ const getAnnotation = (obj: K8sResourceKind, annotation: string): string => {
   return _.get(obj, ['metadata', 'annotations', annotation]);
 };
 
+export const annotationToJSON = (
+  annotationKey: string,
+  annotations: ObjectMetadata['annotations'],
+  onError = (err: Error, key: string) =>
```

Comment:
You don't really need to pass back `key` now because there's only one. I'd change this to

```suggestion
  onError: (err: Error) => void,
```

Then just check if it exists below (maybe always logging the warning).

```ts
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn(`Could not parse annotation "${annotationKey}" as JSON:`, e);
    onError?.(e, annotationKey);
    return undefined;
  }
```

Reply:
My reasoning for creating a default was two fold. It provides implicit typing, and it ensures only one line is logged per error. The method above will log two seperate lines per error (assuming the caller logs the error in their handler), which I think is a little overkill.

Reply:
> My reasoning for creating a default was two fold. It provides implicit typing, and it ensures only one line is logged per error. The method above will log two seperate lines per error (assuming the caller logs the error in their handler), which I think is a little overkill.

I was suggesting to not pass an `onError` and just use the default logging.

I think it's good to always log the warning for debugging. If callers can override, it might get missed.